glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -300,46 +290,75 @@ open class GleanInternalAPI internal constructor () {
      * @param enabled When true, enable metric collection.
      */
     fun setUploadEnabled(enabled: Boolean) {
-        if (isInitialized()) {
-            val originalEnabled = getUploadEnabled()
-
-            @Suppress("EXPERIMENTAL_API_USAGE")
-            Dispatchers.API.launch {
-                LibGleanFFI.INSTANCE.glean_set_upload_enabled(enabled.toByte())
-
-                if (!enabled) {
-                    // Cancel any pending workers here so that we don't accidentally upload or
-                    // collect data after the upload has been disabled.
-                    metricsPingScheduler.cancel()
-                    // Cancel any pending workers here so that we don't accidentally upload
-                    // data after the upload has been disabled.
-                    PingUploadWorker.cancel(applicationContext)
-                }
+        if (!this.initFinished) {
+            val msg = """
+            Changing upload enabled before Glean is initialized is not supported.
+            Pass the correct state into `Glean.initialize()`.
+            See documentation at https://mozilla.github.io/glean/book/user/general-api.html#initializing-the-glean-sdk
+            """.trimIndent()
+            Log.w(LOG_TAG, msg)
```

Comment:
```suggestion
            Log.e(LOG_TAG, msg)
```

Should this be an error?